xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-default/src/main/java/org/xwiki/user/internal/converter/AbstractUserReferenceConverter.java

```diff
@@ -55,6 +61,24 @@ protected UserReference convertToType(Type targetType, Object value)
         return this.userReferenceResolver.resolve(value.toString());
     }
 
+    @Override
+    @SuppressWarnings("unchecked")
+    public <G> G convert(Type targetType, Object sourceValue)
+    {
+        G result;
+
+        // Special handling for target type DocumentReference to support converting UserReference to DocumentReference.
+        if (targetType != null && ReflectionUtils.getTypeClass(targetType).equals(DocumentReference.class)
+            && sourceValue instanceof UserReference)
```

Comment:
It's extremely unlikely that this converter is called if none of the source or the target is a UserReference, but it does not harm I guess.

Reply:
I prefer leaving this as it avoids that the cast in the next line fails.